packages/bsky/src/image/server.ts

```diff
@@ -66,71 +95,133 @@ export class ImageProcessingServer {
 
       // Non-cached flow
 
-      const { localUrl } = this.cfg
-      const did = options.did
-      const cidStr = options.cid.toString()
-
-      const blobResult = await retryHttp(() =>
-        getBlob({ baseUrl: localUrl, did, cid: cidStr }),
-      )
-
-      const imageStream: Readable = blobResult.data
-      const processedImage = await resize(imageStream, options)
-
-      // Cache in the background
-      this.cache
-        .put(cacheKey, cloneStream(processedImage))
-        .catch((err) => log.error(err, 'failed to cache image'))
-      // Respond
-      res.statusCode = 200
-      res.setHeader('x-cache', 'miss')
-      res.setHeader('content-type', getMime(options.format))
-      res.setHeader('cache-control', `public, max-age=31536000`) // 1 year
-      forwardStreamErrors(processedImage, res)
-      return (
-        processedImage
-          // @NOTE sharp does emit this in time to be set as a header
-          .once('info', (info) => res.setHeader('content-length', info.size))
-          .pipe(res)
-      )
-    } catch (err: unknown) {
-      if (err instanceof BadPathError) {
-        return next(createError(400, err))
+      const url = await getBlobUrl(dataplane, did, cid)
+
+      const dispatcherOptions: Dispatcher.RequestOptions = {
+        method: 'GET',
+        origin: url.origin,
+        path: url.pathname + url.search,
+        headers: Object.fromEntries([
+          ...getBlobHeaders(cfg, url),
+          ['accept-encoding', acceptEncoding],
+        ]),
       }
```

Comment:
The implementation gets a little complicated in order to get everything streaming end-to-end.  In case it simplifies the implementation here, this code isn't performance critical and not used in production.  The primary use is to run a self-contained local dev environment which is suitable for UI work, without needing to run an additional service such as imgproxy.